### ISSUE TITLE
Datagrid within tab component with grid does not render correctly in IE 11 

### ISSUE THREAD

**Issue by davidrourke** [vmware/clarity#4035]
## Describe the bug

Datagrid within tab component with grid does not render correctly in IE 11.  I would also like to point out that when you switch tabs in IE, the content is added to the page and does not replace the previous tabs content.

In Chrome:

![image](https://user-images.githubusercontent.com/25104184/69248904-10440e80-0b7b-11ea-9a05-0525fa996d7f.png)

In IE:

![image](https://user-images.githubusercontent.com/25104184/69248951-281b9280-0b7b-11ea-9d95-077875329160.png)

## How to reproduce

I have created a Stackblitz.io project to reproduce the issue:

https://clarity-light-theme-v2-kbd3cb.stackblitz.io

## Expected behavior

The content should render correctly.

## Versions

**App**

* Angular: 8
* Node: 10
* Clarity: 2.0 (and higher)

**Device:**

* Type: MacBook
* OS: 10.15.1
* Browser Chrome, IE 11
* Version Chrome (78.0.3904.97), IE (11.0)

**Comment by davidrourke** [comment]
Any update on when this will be fixed?  I am stuck on v2.1.2 and could really use some of the other fixes in newer release, but I can't because of issue's I've reported with IE 11.

**Comment by gnomeontherun** [comment]
The workaround here is probably to put an NgIf on the datagrid and only flip it to true when the container is rendered, otherwise its trying to calculate a size of its container that might not have been rendered yet.

**Comment by davidrourke** [comment]
I tried this by adding *ngIf to datagrid and enabling in AfterViewInit, but that does not fix the issue.  In addition, the width of the datagrid should fill the entire page, but it does not.  In addition, when you switch between the tabs, the content is being duplicated on the page (only in IE 11), so it does not resolve that issue.

**Comment by hippee-lee** [comment]
@davidrourke - if you can you provide a link to the stackblitz with what you tried i'll take a look at it.

**Comment by davidrourke** [comment]
Matt,

Here is a link:

https://clarity-light-theme-v2-tjgbww.stackblitz.io/ <https://clarity-light-theme-v2-tjgbww.stackblitz.io/>

When you run this in IE 11 besides the grid not sizing correctly, you should also see the tab content get duplicated as switch between the 2 tabs.

Regards,
David

**Comment by davidrourke** [comment]
@gnomeontherun the workaround does not work (as I indicated in my post and example).

**Comment by gnomeontherun** [comment]
Stackblitz doesn't work well in IE11, so this is harder to triage. The issue is a timing issue for rendering the datagrid, so if NgIf doesn't delay it long enough you might need to add a slight timeout as a workaround.

**Comment by davidrourke** [comment]
I upgraded to v2.3.8 to pickup latest changes and then I started debugging on IE 11.  I appears that the width of the DOM section for the tab content is not correct.  I have found a work-around by adding the following to my CSS file:

::ng-deep .tab-content {
     width: 100%
}

**Comment by davidrourke** [comment]
@gnomeontherun, @mathisscott are you aware that the datagrid header is not sticky (i.e. scrolls) in IE 11?  I am trying to create a stackblitz example, but I can't get it to load in IE 11 (which I believe is a known issue).

**Comment by hippee-lee** [comment]
Yes, its a known issue that position: sticky isn't supported in IE11. 
This was a compromise we had to make when refactoring the datagrid in 2018 to open it up for some other features that are in the pipeline. 

If we have have the bandwidth to address this, the fix is to write custom logic for IE11 and handle the positioning of the header with javascript. If I recall correctly none of the polyfills were working when I investigated the sticky approach.
